MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -3669,14 +3725,25 @@ public static string GetHwTonemapFilter(EncodingOptions options, string hwTonema
             if (_mediaEncoder.IsVaapiDeviceInteliHD)
             {
                 // Intel iHD path, with extra vpp tonemap and overlay support.
-                return GetVaapiFullVidFiltersPrefered(state, options, vidDecoder, vidEncoder);
+                return GetIntelVaapiFullVidFiltersPrefered(state, options, vidDecoder, vidEncoder);
+            }
+
+            // prefered vaapi + vulkan filters pipeline
+            if (_mediaEncoder.IsVaapiDeviceAmd
+                && isVaapiVkSupported
+                && _mediaEncoder.IsVaapiDeviceSupportVulkanFmtModifier
+                && Environment.OSVersion.Version != null
```

Comment:
I don't this this can be null?
https://source.dot.net/#System.Private.CoreLib/OperatingSystem.cs,74